model/task/results.go

```diff
@@ -115,3 +115,23 @@ func (c *ResultCounts) String() string { // nolint: golint
 
 	return c.cachedMessage
 }
+
+// FilterTasksOnStatus tasks in a slice of tasks and removes tasks whose result
+// status do not match the passed-in statuses
+func FilterTasksOnStatus(tasks []Task, statuses ...string) []Task {
+	l := len(tasks)
```

Comment:
I might just have a new output slice and append to it rather than do this delete ting.

Reply:
Yea that's a lot cleaner